addons/stock/static/src/js/widgets.js

```diff
@@ -206,39 +206,42 @@ function openerp_picking_widgets(instance){
                 }
             });
             this.$('.js_create_lot').click(function(){
-                var op_id = $(this).parents("[data-id]:first").data('id');
-                var lot_name = false;
                 self.$('.js_lot_scan').val('');
                 var $lot_modal = self.$el.siblings('#js_LotChooseModal');
                 //disconnect scanner to prevent scanning a product in the back while dialog is open
                 self.getParent().barcode_scanner.disconnect();
                 $lot_modal.modal()
-                //focus input
-                $lot_modal.on('shown.bs.modal', function(){
-                    self.$('.js_lot_scan').focus();
-                })
-                //reactivate scanner when dialog close
-                $lot_modal.on('hidden.bs.modal', function(){
-                    self.getParent().barcode_scanner.connect(function(ean){
-                        self.getParent().scan(ean);
-                    });
-                })
                 self.$('.js_lot_scan').focus();
-                //button action
-                self.$('.js_validate_lot').click(function(){
-                    //get content of input
-                    var name = self.$('.js_lot_scan').val();
-                    if (name.length !== 0){
-                        lot_name = name;
-                    }
-                    $lot_modal.modal('hide');
-                    //we need this here since it is not sure the hide event
-                    //will be catch because we refresh the view after the create_lot call
-                    self.getParent().barcode_scanner.connect(function(ean){
-                        self.getParent().scan(ean);
-                    });
-                    self.getParent().create_lot(op_id, lot_name);
-                });
+            });
+            //focus input
+            this.$el.siblings('#js_LotChooseModal').on('shown.bs.modal', function(){
+                self.$('.js_lot_scan').focus();
+            });
+            //reactivate scanner when dialog close
+            this.$el.siblings('#js_LotChooseModal').on('hidden.bs.modal', function(){
+            	self.getParent().barcode_scanner.disconnect(); // remove previous event
+            	self.getParent().barcode_scanner.connect(function(ean){
+            		self.getParent().scan(ean);
+            	});
+            });
+            //button action
+            this.$('.js_validate_lot').click(function(){
+            	//get content of input
+            	var op_id = $("[data-id]:first").data('id');
```

Comment:
this won't work because op_id will always refer to the first line of the operation since the modal is at the root of the DOM. Meaning that if you have 3 different products and you want to add a "lot number" on the second one, it will instead add it to the product on the first line. You can notice that in the previous code, at line 208, the class js_create_lot that opens the modal is inside a product_line, meaning that we could get the id of the line using $(this).parents("[data-id]:first").data('id');

Otherwise it seems correct, though it would be easier and cleaner to just add a function inside the barcode_scanner widget to tell of it is currently connected or not and if we try to connect it while it is already connected, perform nothing instead of registering a new handler.